python/cudf/dataframe/series.py

```diff
@@ -240,6 +240,12 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
         else:
             return NotImplemented
 
+    def __array__(self, dtype=None, **kwargs):
+        out = self.to_array()
+        if dtype and out.dtype != dtype:
+            out = out.astype(dtype)
```

Comment:
Note to self, typecast on the GPU before converting to CPU array as it will be faster.